agreement/types.go

```diff
@@ -93,9 +101,9 @@ const (
 )
 
 func (s step) nextVoteRanges() (lower, upper time.Duration) {
-	extra := recoveryExtraTimeout // eg  2000 ms
-	lower = deadlineTimeout       // eg 17000 ms (15000 + 2000)
-	upper = lower + extra         // eg 19000 ms
+	extra := recoveryExtraTimeout  // eg  2000 ms
+	lower = defaultDeadlineTimeout // eg 17000 ms (15000 + 2000)
+	upper = lower + extra          // eg 19000 ms
```

Comment:
This makes me a little nervous. I suppose it should use the deadlinetimeout for the period it's being invoked in. But then I also don't think anything will go wrong.  This means that the times for all the steps after 1 will not use the short timeout from step 1. I think this just means we'll have a very big time jump after the first two.  So we'll get round of 3, 8, X, instead of today's 3.5, 21, X.

Reply:
That's how I interpreted it, I'm not sure I've seen anything longer than the 21 second rounds to date, but I guess this increases the likelihood of it happening now?

Do you think we should adjust the formula to get something like 3, 8, 21, X?

Reply:
If that's the only effect, I'm not greatly worried about it.  But I'm a little unsure it actually will be the effect. The usage here:
https://github.com/algorand/go-algorand/blob/master/agreement/player.go#L125
seems to be setting a deadline based on the difference between the lower and upper for a step.  If we give the wrong value when in period 0, then I suppose the timeout is wrong. I don't yet know enough to decide if that's a problem (since that timeout is somewhat randomized anyway).

Reply:
I kind of incline `nextVoteRanges` needs to depend on period.
Otherwise on period 0 step 0 has range [17s, 19s], step 1 (in issueSoftVote) has 4s, step 2 has 19s. So it is not monotonic increase and possibly could cause some issue in agreement state machine.
@zeldovich @yossigi opinion?

Reply:
We spoke about this live, hope everyone is now comfortable.

Reply:
This might work out fine (in the sense that it won't stall the Algorand blockchain), but it doesn't seem quite right.  The result of `nextVoteRanges()` is used to set `p.Deadline`, which is how long we wait for the next timeout.  With this PR's change, period 0 might have triggered its previous `DeadlineTimeout` earlier than `defaultDeadlineTimeout`, but this function returns the timeouts based on `defaultDeadlineTimeout`.  As a result, the player will sleep for longer than needed (specifically, the extra difference between `defaultDeadlineTimeout` and the potential shorter period-0 timeout).  Eventually things will almost certainly recover fine, but it seems worth fixing this to be period-specific so that our timeouts are consistent even for period 0.

Reply:
Updated to be period specific!